src/util/ajax.js

```diff
@@ -2,6 +2,38 @@
 
 const window = require('./window');
 
+/**
+ * The type of a resource.
+ */
+const ResourceType = {
+    Unknown: 'Unknown',
+    Style: 'Style',
+    Source: 'Source',
+    Tile: 'Tile',
+    Glyphs: 'Glyphs',
+    SpriteImage: 'SpriteImage',
+    SpriteJSON: 'SpriteJSON',
+    Image: 'Image'
+};
+exports.ResourceType = ResourceType;
+
+if (typeof Object.freeze == 'function') {
+    Object.freeze(ResourceType);
+}
+
+/**
+ * A `RequestParameters` object to be returned from Map.options.transformRequest callbacks.
+ * @typedef {Object} RequestParameters
+ * @property {string} url The URL to be requested.
+ * @property {Object} headers The headers to be sent with the request.
+ * @property {boolean} withCredentials Whether cross-site requests should be made with cookies.
+ */
+export type RequestParameters = {
+    url: string,
+    headers?: Object,
+    withCredentials? : Boolean
```

Comment:
Let's make this a bit more extensible, and consistent with the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch) (which we may eventually switch to internally):

```
    credentials?: 'same-origin' | 'include'
```

(It doesn't look like it's possible to emulate Fetch's `'omit'` value using XHR, so I left that out.)